src/main/java/nl/ordina/jobcrawler/controller/JobcrawlerController.java

```diff
@@ -0,0 +1,80 @@
+package nl.ordina.jobcrawler.controller;
+
+import nl.ordina.jobcrawler.SearchRequest;
+import nl.ordina.jobcrawler.SearchResult;
+import nl.ordina.jobcrawler.model.Vacancy;
+import nl.ordina.jobcrawler.controller.exception.VacancyNotFoundException;
+import nl.ordina.jobcrawler.service.VacancyService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.*;
+
+
+@CrossOrigin(origins = "http://localhost:4200")
+@RestController
+public class JobcrawlerController {
+
+    @Autowired
+    private VacancyService vacancyService;
+
+    // Post mappings
+
+    @PostMapping("/searchrequest")
+    public SearchResult searchRequest(@RequestBody SearchRequest request) {
+
+        // Temporary mock data
+        String requestText = "For request: " + request.getLocation() + ", " + request.getDistance() + ", " + request.getKeywords() +  ":";
```

Comment:
Je kunt overwegen om een toString() methode te overschrijven in SearchRequest. Dat zou deze code een stukje leesbaarder kunnen maken.

Reply:
Has been discussed. Will be changed in the future.